lib/foreman_maintain/concerns/os_facts.rb

```diff
@@ -0,0 +1,68 @@
+module ForemanMaintain
+  module Concerns
+    module OsFacts
+      OS_RELEASE_FILE = '/etc/os-release'.freeze
+      FALLBACK_OS_RELEASE_FILE = '/usr/lib/os-release'.freeze
+
+      def os_release_file
+        if File.file?(OS_RELEASE_FILE)
+          return OS_RELEASE_FILE
+        elsif File.file?(FALLBACK_OS_RELEASE_FILE)
+          return FALLBACK_OS_RELEASE_FILE
+        else
+          puts "The #{OS_RELEASE_FILE} and #{FALLBACK_OS_RELEASE_FILE} files are missing! "\
+               "Can't continue the execution without Operating System's facts!"
+          exit 1
+        end
+      end
+
+      def facts
+        unless defined?(@facts)
+          @facts = {}
+          File.open(os_release_file) do |file|
+            file.readlines.each do |line|
+              line.strip! # drop any whitespace, including newlines from start and end of the line
+              next if line.start_with?('#') # ignore comments
+              # split at most into 2 items, if the value ever contains an =
+              key, value = line.split('=', 2)
+              @facts[key] = value.delete('"') unless key.nil?
```

Comment:
Untested, but perhaps you can use this to support both single and double quotes. I also wondered if we should handle no value at all or imply skip it:
```suggestion
              next unless key && value
              @facts[key] = value.gsub(/^(["'])(.*)(\1)$/, '\2')
```

Reply:
Technically, you'd also need to unquote other chars:
"Variable assignment values must be enclosed in double or single quotes if they include spaces, semicolons or other special characters outside of A–Z, a–z, 0–9. Shell special characters ("$", quotes, backslash, backtick) must be escaped with backslashes, following shell style." (via https://www.freedesktop.org/software/systemd/man/os-release.html)

Reply:
In https://www.freedesktop.org/software/systemd/man/os-release.html#id-1.7.5 they use `ast.literal_eval` in python to parse this properly. No idea if something like this exists in Ruby? (plain eval sounds bad)

Reply:
With the special char and backslash the key value pair comes as `\"\\$bar\"` . The @ekohl regex plucks the string and it comes as `"\\$bar"`, so I think its correct to just do `string.delete("\\"), pushed the change request to check.